components/list/list.js

```diff
@@ -64,13 +71,65 @@ class List extends LitElement {
 		return [layout, specialCases];
 	}
 
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this._checked = {};
+		this.addEventListener('d2l-list-item-checked', this._onlistItemChecked.bind(this));
+	}
+
+	itemsChecked() {
+		return Object.keys(this._checked).filter(ref => this._checked[ref]);
+	}
+
 	render() {
 		return html`
 			<div role="list" class="d2l-list-container">
 				<slot></slot>
 			</div>
 		`;
 	}
+
+	selectionState() {
+		const checkedListItems = this.itemsChecked();
+		if (!checkedListItems || checkedListItems.length < 1) {
+			return selectableListStates.none;
+		}
+
+		const uncheckedListItems = this.querySelectorAll('d2l-list-item:not([checked])');
+		if (uncheckedListItems.length < 1) {
+			return selectableListStates.all;
+		}
+
+		return selectableListStates.indeterminate;
+	}
+
+	selectAll() {
```

Comment:
Would it make more sense to split this into `selectAll()` and `unselectAll()`? It's confusing to me that calling it can do different things depending on how many are checked.

Reply:
Should this be instead `selectToggle`?

Reply:
If you think it's useful as a method, then yeah something like `toggleSelectAll()`?

Reply:
That would work.

Honestly, I am not sure which is better. I think setting up a global checkbox would prefer one method over two. Makes it easier for the caller but this is speculation on my part. 

Does it hurt to have all three? 

- [ ] Change name to `toggleSelectAll`